lib/core/sdam/server.js

```diff
@@ -248,6 +248,61 @@ class Server extends EventEmitter {
     });
   }
 
+  /**
+   * Execute a query against the server
+   *
+   * @param {string} ns The MongoDB fully qualified namespace (ex: db1.collection1)
+   * @param {object} cmd The command document for the query
+   * @param {object} options Optional settings
+   * @param {function} callback
+   */
+  query(ns, cmd, cursorState, options, callback) {
+    wireProtocol.query(this, ns, cmd, cursorState, options, (err, result) => {
+      if (err && isSDAMUnrecoverableError(err)) {
+        this.emit('error', err);
```

Comment:
Why don't we want to emit an error if it's an SDAM recoverable error?

Reply:
Remember, this is what we added for "connections survive primary stepdown"? We only want to emit this error in this case, because it will cause a pool reset

Reply:
oh of course 🤦‍♀